camelot/view/action_steps/orm.py

```diff
@@ -101,27 +99,20 @@ def create_model_context(cls):
 
     @classmethod
     def gui_run(cls, gui_context, serialized_step):
-        step = json.loads(serialized_step)
-        # Presumed to be unnecessary, as ActionStep's gui_run constructs an ActionRunner on it's (empty) model_run,
-        # which results in a unwanted round-trip to the model thread / server.
-        #super(CreateUpdateDelete, self).gui_run(gui_context)
-        crud_signal_handler = get_crud_signal_handler() #CrudSignalHandler()
-        #leases = []
-        if step['deleted'] is not None:
-            crud_signal_handler.objects_deleted.emit(LiveRef(step['deleted']))
-            #crud_signal_handler.objects_deleted.emit(step['deleted'])
-            #leases.append(step['deleted'])
-        if step['updated'] is not None:
-            crud_signal_handler.objects_updated.emit(LiveRef(step['updated']))
-            #crud_signal_handler.objects_updated.emit(step['updated'])
-            #leases.append(step['updated'])
-        if step['created'] is not None:
-            crud_signal_handler.objects_created.emit(LiveRef(step['created']))
-            #crud_signal_handler.objects_created.emit(step['created'])
-            #leases.append(step['created'])
-        #if len(leases):
-            #runner = ActionRunner(unbind_name, cls, leases)
-            #runner.exec()
+        if is_cpp_gui_context(gui_context):
```

Comment:
ik zou verwachten dat deze mss altijd naar cpp mag ?  gezien die crud_signal_handler een cpp object is ?

maar dat is ms siets te simplistisch :)

Reply:
Het probleem is dat we 2 soorten LiveRefs hebben die in leven moeten worden gehouden. Er is een python versie en een C++ versie die een QSharedPointer gebruikt. Ik denk niet dat we die shared pointer kunnen doorgeven aan de CollectionProxy.